### DISCUSSION TITLE
Document usage of private maven repository located via url with credentials.

### DISCUSSION THREAD

**Discussion by rssh** [sbt/sbt#6869] [Q&A] [answered]
In the documentation [https://www.scala-sbt.org/1.x/docs/Resolvers.html], we have nothing about the following case: 
when the maven-based repository, defined by url should be accessed via basic authentication.  Github package registry is a well-known example.

I want to find sbt analog for the following gradle  definition:
```
maven {
        url = uri("https://maven.pkg.github.com/my-organization/my-project")
        credentials {
            username = "my-username"
            password = System.getenv("MY_PROJECT_PASSWORD")
        }
}
```

I found info about credentials, but in credentials, we have the following fields:  (realm, hostname, username, password).  
- But where to put the path? ['organization in path' is different than 'organization in package' ]
- How to make test requests to get realm?  [this is untrivial because requesting the top URL returns us a 404, we need the correct artifact URL for doing this]. 

If we will add a URL repository
```
resolvers += "My repo" at "https://maven.pkg.github.com/my-organization/my-password"

```
The realm will be always  'GitHub Package Registry', and we will not be able to fetch dependencies from more than two repositories.
So, we need some way to link credentials and URLs without the realm.

**Comment by e-orz** [comment, marked as answer]
Hi,
We had the same issue in our company, so I wrote a sbt plugin to handle it. It just maps between a resolver to its credentials. You can create two resolvers, each one with and different path and map each one to its specific credentials.
see:
https://github.com/e-orz/sbt-resolver-credentials
